polar-core/src/inverter.rs

```diff
@@ -169,8 +169,12 @@ impl Runnable for Inverter {
                     return Ok(QueryEvent::Done { result });
                 }
                 QueryEvent::Result { .. } => {
-                    let bindings = self.vm.bindings[self.bsp..].to_owned();
-                    self.results.push(bindings);
+                    if self.vm.query_contains_partial {
+                        let bindings = self.vm.bindings[self.bsp..].to_owned();
+                        self.results.push(bindings);
+                    } else {
+                        return Ok(QueryEvent::Done { result: false });
```

Comment:
What if you have nested nots?

Reply:
Then you'll have nested Inverters -- we have a couple tests for nested `not`s with and without partials.

Reply:
How does this work? The inverter would invert all the constraints already on the partial (from the inner query or from before the inverter runs). I don't think that is correct.

Example:

```
f(x) if
    x = 1 and not (x = 2 or x = 3);
```

Would output with this code (I think):

```
pre invert constraints:

x: partial([x = 1, x = 2, x = 3])

after invert constraints:

x: partial([x != 1 AND x != 2 AND x != 3])
```

Correct result is:

```
x: partial([x = 1 AND (x != 2 AND x != 3)])
```

Reply:
Covered this test case in 8257779 and 68db0ad